src/client/visualizations/base-visualization/base-visualization.tsx

```diff
@@ -109,8 +109,12 @@ export class BaseVisualization<S extends BaseVisualizationState> extends React.C
     return this.debouncedCallExecutor(essence, timekeeper);
   }
 
+  protected getQuery(essence: Essence, timekeeper: Timekeeper): Expression {
```

Comment:
why this function only delegates to another function?

Reply:
This is base/default implementation. Specific Visualisations can override this.

And logic is extracted to function for better maintainability.

Reply:
clear, w need better abstraction for this in the future, just make a note for further work